squidb/src/com/yahoo/squidb/data/SqlValidatorFactory.java

```diff
@@ -0,0 +1,57 @@
+package com.yahoo.squidb.data;
+
+import android.database.Cursor;
+import android.database.sqlite.SQLiteDatabase;
+import android.os.Build.VERSION;
+import android.os.Build.VERSION_CODES;
+
+import com.yahoo.squidb.data.SqlValidatorFactory.SqlValidator;
+
+/*package*/ class SqlValidatorFactory {
+
+    interface SqlValidator {
+
+        void compileStatement(SQLiteDatabase db, String sql);
+    }
+
+    private static final SqlValidator INSTANCE;
+
+    static {
+        int version = VERSION.SDK_INT;
+        if (version >= VERSION_CODES.JELLY_BEAN) {
+            INSTANCE = new DefaultSqlValidator();
+        } else if (version >= VERSION_CODES.ICE_CREAM_SANDWICH) {
+            INSTANCE = new ReflectiveSqlValidator();
```

Comment:
This probably shouldn't be named ReflectiveSqlValidator now that it's no longer using a reflective approach.